core/blockchain.go

```diff
@@ -1827,7 +1837,7 @@ func (bc *BlockChain) reprocessState(current *types.Block, reexec uint64) error
 		// Flatten snapshot if initialized, holding a reference to the state root until the next block
 		// is processed.
 		if err := bc.flattenSnapshot(func() error {
-			if previousRoot != (common.Hash{}) && previousRoot != root {
+			if previousRoot != (common.Hash{}) {
```

Comment:
Are we sure this is ok to not check the root is different from the previous root?